### PR TITLE
GB-5438 - Detect federation

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Detect federated schemas in the CLI rather than passing a flag
- Output the graph type as an enum in SDL to match the parser implementation

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/federated-dev/src/lib.rs [hunk 1]
```diff
@@ -66,11 +66,6 @@ pub fn add_subgraph(name: &str, url: &Url, dev_api_port: u16, headers: Vec<(&str
...
 /// Runs the federated dev system.
-pub fn run(port: u16) -> Result<(), Error> {
-    let runtime = Builder::new_current_thread()
-        .enable_all()
-        .build()
-        .map_err(|error| Error::internal(error.to_string()))?;
-
-    runtime.block_on(dev::run(port))
+pub async fn run(port: u16) -> Result<(), Error> {
+    dev::run(port).await
```
Comment (tomhoule) [inline cli/crates/federated-dev/src/lib.rs:70]: :+1:

File: cli/crates/cli/src/start.rs [hunk 2]
```diff
@@ -85,7 +85,11 @@ async fn log_reporter(mut message_receiver: UnboundedReceiver<ServerMessage>, lo
     while let Some(message) = message_receiver.recv().await {
         #[allow(clippy::single_match)] // will certainly change in the future
         match message {
-            ServerMessage::Ready { listen_address, port } => {
+            ServerMessage::Ready {
+                listen_address,
+                port,
+                is_federated: _,
```
Comment (tomhoule) [inline cli/crates/cli/src/start.rs:91]: I am wondering what happens here if federated is true. Engine v1 can't serve a federated graph, so maybe an error / warning would make sense?

**Review by tomhoule — APPROVED** [review]
Nice and terse :100:

File: cli/crates/cli/src/start.rs (hunk 2 above)
Comment (yoav-lavi) [reply cli/crates/cli/src/start.rs:91]: @tomhoule WDYM? I changed the prod server to also support this

File: cli/crates/federated-dev/src/lib.rs (hunk 1 above)
Comment (tomhoule) [reply cli/crates/federated-dev/src/lib.rs:70]: Removing tokio boilerplate and pushing asyncness to the caller.

File: cli/crates/cli/src/start.rs (hunk 2 above)
Comment (tomhoule) [reply cli/crates/cli/src/start.rs:91]: Aah nice, I hadn't caught on to that, it all makes sense.

**Pull request merged**